src/qibolab/platforms/platform.py

```diff
@@ -0,0 +1,139 @@
+from qibolab.pulses import Pulse, ReadoutPulse
+
+
+class Platform:
+    def __init__(self, design, runcard):
+        self.design = design
+        self.runcard = runcard
+
+        self.native_single_qubit_gates = None
+        self.native_two_qubit_gates = None
+        self.nqubits = None
+        self.resonator_type = None
+        self.topology = None
+        self.qubits = []
+
+        # Load platform settings
+        self.settings = None
+        self.reload_settings()
+
+    def reload_settings(self):
+        """Reloads the runcard and re-setups the connected instruments using the new values."""
+        # TODO: Maybe runcard loading can be improved
+        with open(self.runcard) as file:
+            self.settings = yaml.safe_load(file)
+
+        self.nqubits = self.settings["nqubits"]
+        self.resonator_type = "3D" if self.nqubits == 1 else "2D"
+        self.topology = self.settings["topology"]
+
+        self.native_single_qubit_gates = self.settings["native_gates"].get("single_qubit")
+        self.native_two_qubit_gates = self.settings["native_gates"].get("two_qubit")
+
+        # Create list of qubit objects
+        characterization = self.settings["characterization"]["single_qubit"]
+        self.qubits = []
+        for q, channel_names in self.settings["qubit_channel_map"].items():
+            channels = (Channel(name) for name in channel_names)
+            self.qubits.append(Qubit(q, characterization[q], *channels))
+
+    def connect(self):
+        self.design.connect()
+
+    def setup(self):
+        self.design.setup(self.qubits)
+
+    def start(self):
+        self.design.start()
+
+    def stop(self):
+        self.design.stop()
+
+    def disconnect(self):
+        self.design.disconnect()
+
+    def sweep(self, sequence, *sweepers, nshots=1024):
+        return self.design.sweep(self.qubits, sequence, *sweepers, nshots=nshots)
+
+    def execute_pulse_sequence(self, sequence, nshots=1024):
+        """Play an arbitrary pulse sequence and retrieve feedback.
+
+        Args:
+            sequence (:class:`qibolab.pulses.PulseSequence`): Sequence of pulses to play.
+            nshots (int): Number of hardware repetitions of the execution.
+
+        Returns:
+            TODO: Decide a unified way to return results.
+        """
+        return self.design.play(self.qubits, sequence, nshots)
+
+    # TODO: Maybe channel should be removed from pulses
+    def create_RX90_pulse(self, qubit, start=0, relative_phase=0):
+        kwargs = self.native_single_qubit_gates[qubit]["RX"]
+        kwargs["amplitude"] *= 0.5
+        channel = self.qubits[qubit].drive.name
+        kwargs.update({"start": start, "relative_phase": relative_phase, "channel": channel, "qubit": qubit})
+        return Pulse(**kwargs)
+
+    def create_RX_pulse(self, qubit, start=0, relative_phase=0):
+        qd_duration = self.settings["native_gates"]["single_qubit"][qubit]["RX"]["duration"]
```

Comment:
It would be helpful to have an abstract specs for `settings` as well.

You never know whether `"native_gates"` is in settings or not, so you would fail with `KeyError` at runtime, while a more informative error would be welcome.
A quickfix for the error would be to `try ... except KeyError:` and reraise with a more informative error here, but I'd suggest to go directly for the abstract `Settings` specification.

Reply:
#305